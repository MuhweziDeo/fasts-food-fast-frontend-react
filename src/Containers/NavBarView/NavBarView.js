import React,{ Component } from 'react';
import NavBar from '../../components/Shared/NavBar/NavBar';
const jwt = require('jsonwebtoken');

class NavBarView extends Component{
    state={
        IsAuth:false
    }
    componentWillMount() {
       const  token=localStorage.getItem('token')
        if(token){
            try {
                const decoded= jwt.verify(token, 'thisissesandnkananasckcnakanks');
                this.setState({
                    IsAuth:true
                })
            } catch(err) {
                this.setState({
                    IsAuth:false
                })

            }

        };
    };

    render() {
        return(
            <NavBar IsAuthenticated={this.state.IsAuth}/>
        )
    }
}

export  default NavBarView;
