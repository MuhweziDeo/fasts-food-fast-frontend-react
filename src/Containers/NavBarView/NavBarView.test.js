import React from 'react';
import  { shallow } from 'enzyme';
import  NavBarView  from  './NavBarView';

describe('<NavBarView/>',() => {
    it('should render without crushing', () => {
        let wrapper = shallow(<NavBarView/>)
        expect(wrapper).toMatchSnapshot()

    } );
    it('should render on state change', () => {
        let wrapper = shallow(<NavBarView/>)

        wrapper.setState({
            IsAuth:true
        })
        localStorage.setItem('token','token')
        expect(wrapper).toMatchSnapshot()

    } );
})
