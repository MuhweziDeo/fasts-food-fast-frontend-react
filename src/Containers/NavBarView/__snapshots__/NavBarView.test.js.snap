// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarView/> should render on state change 1`] = `ShallowWrapper {}`;

exports[`<NavBarView/> should render without crushing 1`] = `ShallowWrapper {}`;
